http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -726,16 +722,41 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       task = for {
         _ <- parPreAppCreationSetup
         // create and monitor app
-        _ <- gkeInterp.createAndPollApp(CreateAppParams(msg.appId, msg.project, msg.appName)).adaptError {
-          case e =>
-            PubsubKubernetesError(
-              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.App, None),
-              Some(msg.appId),
-              false,
-              None,
-              None
-            )
-        }
+        _ <- gkeInterp
+          .createAndPollApp(CreateAppParams(msg.appId, msg.project, msg.appName))
+          .onError {
+            case _ =>
+              // clean-up resources in the event of an app creation error
+              for {
+                _ <- logger.info(
+                  s"Attempting to clean up resources due to app creation error for app ${msg.appName} in project ${msg.project}. | trace id: ${ctx.traceId}"
+                )
+                // we need to look up the app because we always want to clean up the nodepool associated with an errored app, even if it was pre-created
+                appOpt <- KubernetesServiceDbQueries.getFullAppByName(msg.project, msg.appId).transaction
+                // note that this will only clean up the disk if it was created as part of this app creation.
+                // it should not clean up the disk if it already existed
+                _ <- appOpt.traverse { app =>
+                  val deleteMsg =
+                    DeleteAppMessage(msg.appId, msg.appName, app.nodepool.id, msg.project, msg.createDisk, msg.traceId)
+                  // This is a good-faith attempt at clean-up. We do not want to take any action if clean-up fails for some reason.
+                  deleteApp(deleteMsg, true).handleErrorWith { e =>
```

Comment:
If we don't want to take any action when clean-up fails, why are we setting `deleteApp`'s `sync` parameter to `true`?

Reply:
This is already being done as an async task, so I think passing `sync = true` to `deleteApp` makes sense here.

Reply:
oh I see